pkgs/by-name/ol/ollama/package.nix

```diff
@@ -52,30 +52,6 @@ let
 
   vendorHash = "sha256-hSxcREAujhvzHVNwnRTfhi0MKI3s8HNavER2VLz6SYk=";
 
-  # ollama's patches of llama.cpp's example server
-  # `ollama/llm/generate/gen_common.sh` -> "apply temporary patches until fix is upstream"
-  # each update, these patches should be synchronized with the contents of `ollama/llm/patches/`
-  llamacppPatches = [
-    (preparePatch "01-load-progress.diff" "sha256-UTmnBS5hQjIL3eXDZc8RBDNJunLlkqJWH20LpXNiGRQ=")
-    (preparePatch "02-clip-log.diff" "sha256-rMWbl3QgrPlhisTeHwD7EnGRJyOhLB4UeS7rqa0tdXM=")
-    (preparePatch "03-load_exception.diff" "sha256-NJkT/k8Mf8HcEMb0XkaLmyUNKV3T+384JRPnmwDI/sk=")
-    (preparePatch "04-metal.diff" "sha256-bPBCfoT3EjZPjWKfCzh0pnCUbM/fGTj37yOaQr+QxQ4=")
-    (preparePatch "05-default-pretokenizer.diff" "sha256-PQ0DgfzycUQ8t6S6/yjsMHHx/nFJ0w8AH6afv5Po89w=")
-    (preparePatch "06-embeddings.diff" "sha256-lqg2SI0OapD9LCoAG6MJW6HIHXEmCTv7P75rE9yq/Mo=")
-    (preparePatch "07-clip-unicode.diff" "sha256-1qMJoXhDewxsqPbmi+/7xILQfGaybZDyXc5eH0winL8=")
-    (preparePatch "09-lora.diff" "sha256-tNtI3WHHjBq+PJZGJCBsXHa15dlNJeJm+IiaUbFC0LE=")
-    (preparePatch "11-phi3-sliding-window.diff" "sha256-VbcR4SLa9UXoh8Jq/bPVBerxfg68JZyWALRs7fz7hEs=")
-  ];
-
-  preparePatch =
-    patch: hash:
-    fetchpatch {
-      url = "file://${src}/llm/patches/${patch}";
-      inherit hash;
-      stripLen = 1;
-      extraPrefix = "llm/llama.cpp/";
-    };
-
```

Comment:
It *might* be worth putting the patch hook as `prePatch` or `postUnpack` rather than `postPatch` if we expect to apply patches of our own, or even in the `postFetch` of the src derivation, so that we can treat the loose patches as just an unfortunate distribution detail of the upstream source.

Reply:
If you want to move the contents of `postPatch` to `postFetch`, I'd be totally fine with that; I can't think of any reason not to.

> Sticking it in postFetch would have the tiny bonus of not applying them in both the module fetch and the actual build

I don't think I understand: are you saying the patches are currently being applied twice? If so, please elaborate.

Reply:
> I don't think I understand: are you saying the patches are currently being applied twice? If so, please elaborate.

There are two derivations: the Go module fetch one, which builds the vendor directory, and the actual build.
`patches` and `postPatch` are applied in both. The overhead is pretty negligible though.